src/devices/gridstream.c

```diff
@@ -0,0 +1,258 @@
+                    if (crc16(&b[6], stream_len - 2, 0x1021, known_crc_init[crc_count]) == crc) {
+                        crc_ok = true;
+                    }
+                    else {
+                        crc_count++;
+                    }
+                } while (crc_count < (sizeof(known_crc_init) / 2) && crc_ok == false);
+                if (!crc_ok) {
+                    decoder_log(decoder, 1, __func__, "Bad CRC or unknown init value. ");
+                    if (stream_len == 0x23) {
+                        decoder_log_bitrow(decoder, 1, __func__, &b[6], decoded_len * 8, "Use RevEng to find init value.");
+                    }
+                    return DECODE_FAIL_MIC;
+                }
+
+                sprintf(destwanaddress_str, "%02x%02x%02x%02x%02x%02x", b[7], b[8], b[9], b[10], b[11], b[12]);
+                sprintf(srcwanaddress_str, "%02x%02x%02x%02x%02x%02x", b[13], b[14], b[15], b[16], b[17], b[18]);
+                sprintf(srcaddress_str, "%02x%02x%02x%02x", b[26], b[27], b[28], b[29]);
+                uptime = ((uint32_t)b[20] << 24) | (b[21] << 16) | (b[22] << 8) | b[23];
+
+                /* clang-format off */
+                        data = data_make(
+                                "model",       "", DATA_STRING,  "LandisGyr GridStream",
+                                "id",          "Source Meter ID", DATA_STRING,  srcaddress_str,
+                                "subtype",     "", DATA_INT,     subtype,
+                                "wanaddress",  "", DATA_STRING,  srcwanaddress_str,
+                                "destaddress", "", DATA_STRING,  destwanaddress_str,
+                                "uptime",      "", DATA_INT,     uptime,
+                                "mic",         "Integrity", DATA_STRING,  "CRC", // CRC, CHECKSUM, or PARITY
+                                NULL);
+                /* clang-format on */
+
+                break;
+            case 0xD2:
+                stream_len = b[4];
+                if ((decoded_len - 5) < stream_len) {
+                    return DECODE_ABORT_LENGTH;
+                }
+                crc = (b[3 + stream_len] << 8) | b[4 + stream_len];
+                do {
+                    if (crc16(&b[5], stream_len - 2, 0x1021, known_crc_init[crc_count]) == crc) {
+                        crc_ok = true;
+                    }
+                    else {
+                        crc_count++;
+                    }
+                } while (crc_count < (sizeof(known_crc_init) / 2) && crc_ok == false);
+                if (!crc_ok) {
+                    decoder_log(decoder, 1, __func__, "Bad CRC or unknown init value. ");
+                    return DECODE_FAIL_MIC;
+                }
+
+                /* clang-format off */
+                        data = data_make(
+                                "model",      "", DATA_STRING, "LandisGyr GridStream",
+                                "id",         "", DATA_STRING,    "n/a",
```

Comment:
Not sure we should output this key.

Reply:
I was not certain either. It was not referenced in the wiki, but it is referenced in the gnuradio gridstream decoder. I have seen these broadcasted, but I'm not certain if the timing is different or something else, because I have not seen any pass the CRC check. None of the data is obvious like the others.

Reply:
We if data is present in the bitstream then we should output it. In this case it is hard coded to "n/a".

Reply:
All of the data is unknown in this subtype. Without a captured packet stream that passes CRC, it is hard to go further. I'm okay with either dumping the decoded frame, provided it passes CRC, or just omitting it altogether.

Reply:
I want the complete frame dumped in all cases as this will aid RE work.

Reply:
Would it be appropriate to convert it to char with bitrow_asprint_code and output under data, or just use decoder_log_bitrow with level 0?

Reply:
Do not print anything, it has to be `decoder_log_` so we can redirect to e.g. a web UI.

Reply:
Addressed in https://github.com/merbanan/rtl_433/pull/2616/commits/7edd93a5aead3c669f4bd8e5dbf2d6d32e9ede96